doc/lua-filters.md

```diff
@@ -2021,6 +2023,121 @@ Fields:
 :   track changes setting for docx; one of `accept-changes`,
     `reject-changes`, and `all-changes` (string)
 
+## WriterOptions {#type-writeroptions}
+
+Pandoc writer options
+
+Fields:
+
+`cite_method`
+:   How to print cites -- one of 'citeproc', 'natbib', or
+    'biblatex' (string)
+
+`columns`
+:   Characters in a line (for text wrapping) (integer)
+
+`dpi`
+:   DPI for pixel to/from inch/cm conversions (integer)
+
+`email_obfuscation`
+:   How to obfuscate emails -- one of 'none', 'references', or
+    'javascript' (string)
+
+`epub_chapter_level`
+:   Header level for chapters, i.e., how the document is split
+    into separate files (integer)
+
+`epub_fonts`
+:   Paths to fonts to embed (sequence of strings)
+
+`epub_metadata`
+:   Metadata to include in EPUB (string|nil)
+
+`epub_subdirectory`
+:   Subdir for epub in OCF (string)
+
+`extensions`
+:   Markdown extensions that can be used (sequence of strings)
+
+`highlight_style`
+:   Style to use for highlighting; see the output of `pandoc
+    --print-highlight-style=...` for an example structure. The
+    value `nil` means that no highlighting is used. (table|nil)
+
+`html_math_method`
+:   How to print math in HTML; one 'plain', 'gladtex', 'webtex',
+    'mathml', 'mathjax', or a table with keys `method` and
+    `url`. (string|table)
+
+`html_q_tags`
+:   Use `<q>` tags for quotes in HTML (boolean)
+
+`identifier_prefix`
+:   Prefix for section & note ids in HTML and for footnote marks
+    in markdown (string)
+
+`incremental`
+:   True if lists should be incremental (boolean)
+
+`listings`
+:   Use listings package for code (boolean)
+
+`number_offset`
+:   Starting number for section, subsection, ... (sequence of
+    integers)
+
+`number_sections`
+:   Number sections in LaTeX (boolean)
+
+`prefer_ascii`
+:   Prefer ASCII representations of characters when possible
+    (boolean)
+
+`reference_doc`
+:   Path to reference document if specified (string|nil)
+
+`reference_links`
+:   Use reference links in writing markdown, rst (boolean)
+
+`reference_location`
+:   Location of footnotes and references for writing markdown;
+    one of 'end-of-block', 'end-of-section', 'end-of-document'.
+    The common prefix may be omitted when setting this value.
+    (string)
+
+`section_divs`
+:   Put sections in div tags in HTML (boolean)
+
+`setext_headers`
+:   Use setext headers for levels 1-2 in markdown (boolean)
+
+`slide_level`
+:   Force header level of slides (integer\|nil)
+
+`tab_stop`
+:   Tabstop for conversion btw spaces and tabs (integer)
+
+`table_of_contents`
+:   Include table of contents (boolean)
+
+`toc_depth`
+:   Number of levels to include in TOC (integer)
+
+`top_level_division`
+:   Type of top-level divisions; one of 'top-level-part',
+    'top-level-chapter', 'top-level-section', or
+    'top-level-default'. The prefix `top-level` may be omitted
+    when setting this value. (string)
+
+`variables`
+:   Variables to set in template; string-indexed table (table)
```

Comment:
I guess `variables` is pretty useless unless you can specify a template.
How hard would it be to wrap `compileTemplate`?

Reply:
Not hard. We'd need a (opaque) userdata type for `Template Text` and a fairly short function definition of maybe 6 lines of code. Were would that function live? `pandoc.compile_template` or maybe `pandoc.template.compile`?

Reply:
I've added two commits, providing `pandoc.compile_template`.  I'd probably prefer `pandoc.template.compile` better, just in case we decide to expose more template-related functions.

Reply:
I don't have strong feelings about the naming issue.  I think the other function that would be useful is getDefaultTemplate.

Reply:
I created a `pandoc.template` module with two functions, `compile` to compile a template, and `default` to retrieve the default template for a writer. I left out `compileDefaultTemplate` as it can be built from the other two.

Reply:
>  left out compileDefaultTemplate as it can be built from the other two.

But compileDefaultTemplate handles the partials for you; that might be complicated to reproduce with lua?

Reply:
I believe the following should do, but I might have missed something:

``` lua
function compile_default_template (writer)
  return pandoc.template.compile(pandoc.template.default(writer))
end
```

See also the last test in file `pandoc-template.lua`.

Reply:
I should add that `compile` now uses `runWithDefaultPartials` if no template path is given.

Reply:
OK, if you've tested it with partials, I'm satisfied.